### PR TITLE
Set openapi schema type to file where possible

### PR DESCRIPTION
**Pull request by crobibero**
Swashbuckle doesn't support generating `Schema Type: File`, so have to create a filter that will edit the schema.
AspNetCore doesn't support `[Produces()]` with a wildcard.

#4029

### CODE REVIEW

File: Jellyfin.Api/Controllers/DashboardController.cs [hunk 1]
```diff
@@ -211,9 +214,8 @@ public ActionResult GetWebClientResource([FromRoute] string resourceName)
             {
                 return Redirect("index.html?start=wizard#!/wizardstart.html");
             }
-
-            var stream = new FileStream(_resourceFileManager.GetResourcePath(basePath, path), FileMode.Open, FileAccess.Read);
-            return File(stream, MimeTypes.GetMimeType(path));
+
```
Comment (nielsvanvelzen) [inline Jellyfin.Api/Controllers/DashboardController.cs:217]: Getting build errors on this line
`  DashboardController.cs(217, 1): [SA1028] Code should not contain trailing whitespace`

**Comment by nielsvanvelzen** [comment]
Still missing a return type:
- GetItemImage2

**Comment by nielsvanvelzen** [comment]
Unfortunately you managed to break a few other operations now, all from ImageByNameControl:

```
Missing return-type for operation GetGeneralImage (status-codes: [200, 404])
Missing return-type for operation GetMediaInfoImage (status-codes: [200, 404])
Missing return-type for operation GetRatingImage (status-codes: [200, 404])
```

**Review by Ullmie02 — APPROVED** [review]

**Review by Camc314 — APPROVED** [review]

File: Jellyfin.Api/Controllers/ItemLookupController.cs [hunk 2]
```diff
@@ -274,10 +277,8 @@ public async Task<ActionResult<IEnumerable<RemoteSearchResult>>> GetBookRemoteSe
             }
 
             await DownloadImage(providerName, imageUrl, urlHash, pointerCachePath).ConfigureAwait(false);
-
-            // Read the pointer file again
-            await using var fileStream = System.IO.File.OpenRead(pointerCachePath);
-            return new FileStreamResult(fileStream, MediaTypeNames.Application.Octet);
+            var updatedContentPath = await System.IO.File.ReadAllTextAsync(pointerCachePath).ConfigureAwait(false);
```
Comment (cvium) [inline Jellyfin.Api/Controllers/ItemLookupController.cs:280]: Why are you reading the file?

File: Jellyfin.Api/Controllers/SystemController.cs [hunk 3]
```diff
@@ -190,16 +192,13 @@ public ActionResult<EndPointInfo> GetEndpointInfo()
...
         [Authorize(Policy = Policies.RequiresElevation)]
         [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesFile(MediaTypeNames.Text.Plain)]
         public ActionResult GetLogFile([FromQuery, Required] string? name)
         {
             var file = _fileSystem.GetFiles(_appPaths.LogDirectoryPath)
                 .First(i => string.Equals(i.Name, name, StringComparison.OrdinalIgnoreCase));
 
-            // For older files, assume fully static
-            var fileShare = file.LastWriteTimeUtc < DateTime.UtcNow.AddHours(-1) ? FileShare.Read : FileShare.ReadWrite;
```
Comment (cvium) [inline Jellyfin.Api/Controllers/SystemController.cs:199]: Is this irrelevant now?

File: Jellyfin.Api/Controllers/VideoAttachmentsController.cs [hunk 4]
```diff
@@ -1,5 +1,6 @@
 using System;
 using System.ComponentModel.DataAnnotations;
+using System.IO;
```
Comment (cvium) [inline Jellyfin.Api/Controllers/VideoAttachmentsController.cs:3]: ?

File: Jellyfin.Api/Controllers/ItemLookupController.cs (hunk 2 above)
Comment (crobibero) [reply Jellyfin.Api/Controllers/ItemLookupController.cs:280]: https://github.com/jellyfin/jellyfin/blob/b49cd1d3017f23fc75703829ac2ea1d45d8a4881/MediaBrowser.Api/ItemLookupService.cs#L285

Looks like this was migrated from the servicestack api

**Review by cvium — APPROVED** [review]

**Pull request merged**